beacon-chain/light-client-network.md

```diff
@@ -0,0 +1,200 @@
+[LightClientStore](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientstore) with the initial sync committee and block header from a user-configured trusted block root.
+
+Once the client establishes a recent header, it could sync to other headers by processing objects of type [LightClientUpdate](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientupdate),
+[LightClientFinalityUpdate](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientfinalityupdate)
+and [LightClientOptimisticUpdate](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientoptimisticupdate).
+These data types allow a client to stay up-to-date with the beacon chain.
+
+The Beacon Chain Light Client network is a [Kademlia](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) DHT that forms an overlay network on top of
+the [Discovery v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md) network. The term *overlay network* means that the light client network operates
+with its routing table independent of the base Discovery v5 routing table and uses the extensible `TALKREQ` and `TALKRESP` messages from the base Discovery v5 protocol for communication.
+
+The `TALKREQ` and `TALKRESP` protocol messages are application-level messages whose contents are specific to the Beacon Chain Light Client network. We specify these messages below.
+
+The Beacon Chain Light Client network uses a modified version of the routing table structure from the Discovery v5 network and the lookup algorithm from section 2.3 of the Kademlia paper.
+
+### Gossip Algorithm
+
+A gossip network allows participants to receive regular updates on a particular data type. The key point of differentiation between a portal network gossip
+channel and a regular gossip channel, e.g. the gossip topic `beacon_block` used by regular beacon chain clients, is that a portal network participant could choose an <em>interest radius</em> `r`.
+Participants only process messages that are within their chosen radius boundary.
+
+- Each gossip participant has a `node_id`. A `node_id` is a 256 bit unsigned integer, i.e. `0 <= node_id < 2**256`.
+- There is a distance function. It measures the distance between two `node_id`. It also measures the distance between `node_id` and `content_id`.
+
+#### Validation
+Validating `LightClientFinalityUpdate` and `LightClientOptimisticUpdate` follows the gossip domain(gossipsub) [consensus specs](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#the-gossip-domain-gossipsub).
+
+### Data
+
+#### Types
+
+* LightClientBootstrap
+* LightClientUpdate
+* LightClientFinalityUpdate
```

Comment:
As I understand both `LightClientFinalityUpdate` and `LightClientOptimisticUpdate` are ephemeral objects which are created by full nodes and gossiped to interested light nodes over gossipbupsub so that those can stay in sync.  (or am i missing something here ?)

What is main use case for storing them ? (I kind of assume we need to store them to respond queries)

Reply:
> What is main use case for storing them ? (I kind of assume we need to store them to respond queries)

Yes, those objects are ephemeral and I'm not sure if we need to store them (probably only storing the recent updates)